src/Compilers/Core/Portable/CodeAnalysisResources.resx

```diff
@@ -363,6 +363,9 @@
   <data name="CompilerAnalyzerThrows" xml:space="preserve">
     <value>The Compiler Analyzer '{0}' threw an exception with message '{1}'.</value>
   </data>
+  <data name="DiagnosticDescriptorThrows" xml:space="preserve">
+    <value>The DiagnosticDescriptor with ID '{0}' threw an exception with message '{1}'.</value>
```

Comment:
Since this is a user-facing error message, talking about the descriptor sounds weird. Can we just say "An error was encountered while producing a diagnostic with Id {0} - {1}"?